File: backend/app/schemas/order.py
"""订单相关的请求/响应模型。"""

from datetime import datetime
from decimal import Decimal
from typing import Literal

from pydantic import BaseModel, Field

from app.schemas.common import PageResponse


class OrderCreate(BaseModel):
    receiver_name: str = Field(min_length=1, max_length=50)
    receiver_phone: str = Field(min_length=1, max_length=20)
    receiver_address: str = Field(min_length=1, max_length=200)


class OrderItemResponse(BaseModel):
    id: int
    product_id: int
    sku_id: int | None
    product_name: str
    price: Decimal
    quantity: int
    subtotal: Decimal

    model_config = {"from_attributes": True}


class OrderResponse(BaseModel):
    id: int
    order_no: str
    status: str
    total_amount: Decimal
    username: str | None = None
    receiver_name: str
    receiver_phone: str
    receiver_address: str
    created_at: datetime
    items: list[OrderItemResponse]

    model_config = {"from_attributes": True}


class OrderListResponse(PageResponse[OrderResponse]):
    """订单分页响应（字段与前端 OrderListResult 一致）。"""


class OrderStatusUpdate(BaseModel):
    status: Literal["pending", "paid", "shipped", "completed", "cancelled"]

File: frontend/src/types/order.ts
export type OrderStatus = 'pending' | 'paid' | 'shipped' | 'completed' | 'cancelled'

export interface OrderItem {
  id: number
  product_id: number
  sku_id: number | null
  product_name: string
  price: string
  quantity: number
  subtotal: string
}

export interface Order {
  id: number
  order_no: string
  status: OrderStatus
  total_amount: string
  username: string | null
  receiver_name: string
  receiver_phone: string
  receiver_address: string
  created_at: string
  items: OrderItem[]
}

export interface OrderListResult {
  items: Order[]
  total: number
  page: number
  page_size: number
}

export interface OrderPayload {
  receiver_name: string
  receiver_phone: string
  receiver_address: string
}

export const ORDER_STATUS_TEXT: Record<OrderStatus, string> = {
  pending: '待支付',
  paid: '已支付',
  shipped: '已发货',
  completed: '已完成',
  cancelled: '已取消',
}

export const ORDER_STATUS_TAG: Record<OrderStatus, 'warning' | 'primary' | 'info' | 'success' | 'danger'> = {
  pending: 'warning',
  paid: 'primary',
  shipped: 'info',
  completed: 'success',
  cancelled: 'danger',
}
